extensions/amp-wistia-player/0.1/amp-wistia-player.js

```diff
@@ -0,0 +1,211 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  removeElement,
+  fullscreenEnter,
+  fullscreenExit,
+  isFullscreenElement,
+} from '../../../src/dom';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {user, dev} from '../../../src/log';
+import {
+  installVideoManagerForDoc,
+} from '../../../src/service/video-manager-impl';
+import {VideoEvents} from '../../../src/video-interface';
+import {Services} from '../../../src/services';
+
+/**
+ * @implements {../../../src/video-interface.VideoInterface}
+ */
+class AmpWistiaPlayer extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+    /** @private {?Element} */
+    this.iframe_ = null;
+  }
+
+  /**
+   * @param {boolean=} onLayout
+   * @override
+   */
+  preconnectCallback(onLayout) {
+    // video player and video metadata
+    this.preconnect.url('https://fast.wistia.net', onLayout);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+
+  /** @override */
+  buildCallback() {
+    installVideoManagerForDoc(this.element);
+    Services.videoManagerForDoc(this.element).register(this);
+  }
+
+  /** @override */
+  layoutCallback() {
+    const mediaId = user().assert(
+        this.element.getAttribute('data-media-id'),
+        'The data-media-id attribute is required for <amp-wistia-player> %s',
+        this.element);
+    const iframe = this.element.ownerDocument.createElement('iframe');
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('allowfullscreen', 'true');
+    iframe.src = '//fast.wistia.net/embed/iframe/' + encodeURIComponent(
+        mediaId);
+    this.applyFillContent(iframe);
+    this.element.appendChild(iframe);
+    this.iframe_ = iframe;
+    return this.loadPromise(iframe);
+  }
+
+  /** @override */
+  unlayoutCallback() {
+    if (this.iframe_) {
+      removeElement(this.iframe_);
+      this.iframe_ = null;
+    }
+
+    return true;
+  }
+
+  /** @override */
+  pauseCallback() {
+    if (this.iframe_) {
+      this.pause()
+    }
+  }
+
+  /**
+   * Sends a command to the player through postMessage.
+   * @param {string} command
+   * @private
+   */
+  sendCommand_(command) {
+    if (this.iframe_ && this.iframe_.contentWindow) {
+      this.iframe_.contentWindow./*OK*/postMessage(command, '*');
```

Comment:
Any idea what that `/*OK*/` does?

Reply:
it's a comment; shouldn't do anything. I think it's just a note just to remind ourselves that it's authorized cross-site behaviour.

Reply:
Yeah, I guess I was just wondering if it's some compilation macro or something. Weird, haha.